src/main/java/io/minio/MinioClient.java

```diff
@@ -152,7 +152,7 @@
    * @see #MinioClient(String endpoint, int port, String accessKey, String secretKey, boolean secure)
    */
   public MinioClient(String endpoint) throws InvalidEndpointException, InvalidPortException {
-    this(endpoint, 0, null, null, true);
+    this(endpoint, 0, null, null, endpoint==null?false:endpoint.startsWith("https://"));
```

Comment:
@sdkrum i have fixed those and sent a pull request https://github.com/UNINETT/minio-java/issues/1.  I guess you haven't seen my private message in gitter.  Can you please check whether that is ok?